### PR TITLE
Destination postgres: add option for cascade drop

### LINKED ISSUE #35386: [destination-postgres] Dv2 does not use CASCADE when deleting tables
### Connector Name

destination-postgres

### Connector Version

2.0.0

### What step the error happened?

During the sync

### Relevant information

Hello, it appears that there are some cascades missing from drop queries when running resets with postgres dest v 2.0.0

### Relevant log output

```shell
java.lang.RuntimeException: org.postgresql.util.PSQLException: ERROR: cannot drop table schema_name.table_name because other objects depend on it Detail: view schema_name depends on table schema_name.table_name Hint: Use DROP ... CASCADE to drop the dependent objects too.

```
```


### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte/issues/35386

* AbstractJdbcDestinations.getSqlGenerator now accepts `config: JsonNode`
* JdbcSqlGenerator now accepts `cascadeDrop: Boolean = false`, and adds `cascade()` to its drop table statements
  * there are three of these; I only modified two of them (the ones that drop the final table). The v1v2 migration drops the v2 raw table if it already exists, which IMO if we need to cascade drop, then something is wrong. The v1v2 migration shouldn't even be triggering if the user is already building stuff on top of v2 tables, because it should only run if the v2 tables don't exist.
* Add a testcase to the postgres sqlgenerator integration test to verify cascade works (tested locally to verify that the old code fails the test)

I didn't do any manual syncs/etc, lmk if you want that, or e.g. want a screenshot of the config option, etc

### CODE REVIEW

**Review by gisripa — APPROVED** [review]
lgtm. 

Add a doc update in postgres.md with big `:::caution` 😅

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt [hunk 1]
```diff
@@ -285,25 +292,31 @@ abstract class JdbcSqlGenerator(protected val namingTransformer: NamingConventio
...
     }
 
     override fun overwriteFinalTable(stream: StreamId, finalSuffix: String): Sql {
+        val dropTableStep = DSL.dropTableIfExists(DSL.name(stream.finalNamespace, stream.finalName))
+        val dropTableSql =
+            if (cascadeDrop) {
+                dropTableStep.cascade().getSQL(ParamType.INLINED)
+            } else {
+                dropTableStep.getSQL(ParamType.INLINED)
+            }
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt:317]: kotlin expression ? 
```
dropTableSql = if (cascade) .. else ..
```

Comment (edgao) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt:317]: it's already an expression, the autoformatter just moved it to a new line. I'm happy to remove the braces, no strong format preference about that. autoformatter produces this:
```kotlin
        val dropTableSql =
            if (cascadeDrop) dropTableStep.cascade().getSQL(ParamType.INLINED)
            else dropTableStep.getSQL(ParamType.INLINED)
```

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt [hunk 2]
```diff
@@ -265,13 +265,20 @@ abstract class JdbcSqlGenerator(protected val namingTransformer: NamingConventio
...
+
+        val dropTableStep =
+            DSL.dropTableIfExists(DSL.quotedName(stream.id.finalNamespace, finalTableIdentifier))
+        val dropTableSql =
+            if (cascadeDrop) {
+                dropTableStep.cascade().getSQL(ParamType.INLINED)
+            } else {
+                dropTableStep.getSQL(ParamType.INLINED)
+            }
+
```
Comment (jbfbell) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt:277]: nitpick:
```suggestion

        val dropTableStep =
            DSL.dropTableIfExists(DSL.quotedName(stream.id.finalNamespace, finalTableIdentifier))
        if (cascadeDrop) {
            dropTableStep.cascade()
        } 
        val dropTableSql = dropTableStep.getSQL(ParamType.INLINED)

```

and the `dropTableSql` param is probably unnecessary

**Review by jbfbell — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt (hunk 2 above)
Comment (edgao) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.kt:277]: ... I did not realize jooq objects actually mutate themselves, had assumed they returned a copy. neat!

**Pull request merged**